packages/flutter/lib/src/material/switch.dart

```diff
@@ -758,13 +808,17 @@ class _RenderSwitch extends RenderToggleable {
         break;
     }
 
-    final Color trackColor = isEnabled
+    final Color trackColor = isEnabled || useActiveColorInDisabledState
       ? Color.lerp(inactiveTrackColor, activeTrackColor, currentValue)!
       : inactiveTrackColor;
 
-    final Color thumbColor = isEnabled
+    final Color lerpedThumbColor = isEnabled || useActiveColorInDisabledState
       ? Color.lerp(inactiveColor, activeColor, currentValue)!
       : inactiveColor;
+    // Blend the thumb color against a `surfaceColor` background in case the
+    // thumbColor is not opaque. This way we do not see through the thumb to the
+    // track underneath.
+    final Color thumbColor = Color.alphaBlend(lerpedThumbColor, surfaceColor);
```

Comment:
tests look great and thorough, but I couldn't find a place where this was tested. Can you add an expect or a comment where applicable?

Reply:
Good call, added a test.